### ISSUE TITLE
current mock incompatible with rpm-ostree 2016.7 which uses bubblewrap

### ISSUE THREAD

**Issue by cgwalters** [fedora-infra/fedmsg-atomic-composer#4]
In Fedora 24, rpm-ostree detected whether or not it could use container features itself, and fell back to not doing so.  This mostly worked.

In Fedora 25, dracut now fails to add any kernel modules if there's no `/proc/modules`; see https://bugzilla.redhat.com/show_bug.cgi?id=1374082#c4
And in general, to be correct rpm-ostree really needs to use container features itself, for the same reasons mock does.

There's a fixed rpm-ostree which works "bare" (virt/metal), as well as inside a privileged Docker container.  I also verified the code works fine inside systemd-nspawn (e.g. `mock --new-chroot`).  However, it fails in `mock` (without `--new-chroot`) which is what this project uses.

I think we should try to get `mock --new-chroot` to work, but we were hitting some bugs (related to `mock --clean`), and others.

**Comment by kushaldas** [comment]
In a F24 cloud vm (updated), I can see `/proc/modules` even without `--new-chroot`. But the dracut is still failing to find it. 

```
    libkmod: kmod_module_new_from_loaded: could not open /proc/modules: No such file or directory
    dracut-install: Could not get list of loaded modules: Unknown error -2
```

**Comment by cgwalters** [comment]
That's because rpm-ostree runs dracut (chrooted before 2016.7, proper bubblewrap after) inside its own just-built root.

**Comment by lmacken** [comment]
the fedmsg-atomic-composer /should/ be doing `mock --new-chroot` on rhel7+... 

https://github.com/fedora-infra/fedmsg-atomic-composer/blob/develop/fedmsg_atomic_composer/config.py#L118

However, looking at the `bodhi-backend02:/srv/fedora-atomic/logs/`, I don't think that is getting used. More investigation is needed.

**Comment by cgwalters** [comment]
@lmacken It was disabled in https://infrastructure.fedoraproject.org/cgit/ansible.git/commit/?id=32ea396c4be7fa5732b56e0e503efacc1634b410

But unfortunately the commit message doesn't say why.  My guess here is the old internal containerization broke in nspawn, but now it (or Docker) would be preferable since it's pretty well known how to do recursive containerization with those.

**Comment by cgwalters** [comment]
So...in my testing, `rpm-ostree-2016.7` works just fine inside `mock --new-chroot` with `mock-1.2.20-1.fc24.noarch`.  Maybe different mock versions are in play here?

**Comment by nirik** [comment]
This is fixed with PR#5. 

I've hotfixed our instance, but we will at some point soon want to do a new release with these changes and drop the hotfix.
